summingbird-storm/src/main/scala/com/twitter/summingbird/storm/StormPlatform.scala

```diff
@@ -184,19 +187,17 @@ abstract class Storm(options: Map[String, Options], transformConfig: Summingbird
 
     val metrics = getOrElse(stormDag, node, DEFAULT_SPOUT_STORM_METRICS)
 
-    val registerAllMetrics = new Function1[TopologyContext, Unit] {
-      def apply(context: TopologyContext) = {
-        // Register metrics passed in SpoutStormMetrics option.
-        metrics.metrics().foreach {
-          x: StormMetric[IMetric] =>
-            context.registerMetric(x.name, x.metric, x.interval.inSeconds)
-        }
-        // Register summingbird counter metrics.
-        StormStatProvider.registerMetrics(jobID, context, countersForSpout)
-        SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
+    val registerAllMetrics = Externalizer({ context: TopologyContext =>
+      // Register metrics passed in SpoutStormMetrics option.
+      metrics.metrics().foreach {
+        x: StormMetric[IMetric] =>
+          context.registerMetric(x.name, x.metric, x.interval.inSeconds)
       }
-    }
-    val stormSpout = tormentaSpout.openHook(registerAllMetrics).getSpout
+      // Register summingbird counter metrics.
+      StormStatProvider.registerMetrics(jobID, context, countersForSpout)
+      SummingbirdRuntimeStats.addPlatformStatProvider(StormStatProvider)
+    })
+    val stormSpout = tormentaSpout.openHook(registerAllMetrics.get).getSpout
```

Comment:
how does this help if we are just calling `.get`? Are you sure this is needed? Would marking it `@transient` also work instead (and be simpler)?